1732Sprint/3-Sheik/Chapter6/src/main/java/Exceptions.java

```diff
@@ -36,30 +41,36 @@ private void checkCode(String input) {
     private void checkLength(String[] inputCode) {
         for (int i = 0; i < inputCode.length; i++) {
             try {
-                validateInput(inputCode[i], CODE[i], i);
+                validateInput(inputCode[i], Integer.parseInt(CODE[i]));
             } catch (ArrayIndexOutOfBoundsException e) {
                 System.out.println(i + ":");
                 e.printStackTrace();
             }
         }
     }
 
-    private void validateInput(String inputCode, String code, int digit) {
-        try {
-            Integer.parseInt(inputCode);
-            Integer.parseInt(code);
+    private void validateInput(String inputCode, int code) {
+        int parsedInput = 0;
 
-            if (inputCode.equals(code)) {
-                System.out.println(digit + ": Digit correct");
-            } else {
-                System.out.println(digit + ": Digit incorrect");
-            }
+        try {
+            parsedInput = Integer.parseInt(inputCode);
         } catch (NumberFormatException e) {
-            if (inputCode.equals(code)) {
-                System.out.println(digit + ": Letter correct");
-            } else {
-                System.out.println(digit + ": Letter incorrect");
-            }
+            e.printStackTrace();
+        }
+
+        if (parsedInput == code) {
+            System.out.println(parsedInput + " correct");
+        } else {
+            System.out.println(parsedInput + " incorrect");
+        }
+    }
+
+    private void codeCracked(String input) {
+        if (input.equals("1234")) {
```

Comment:
Maybe it's better to put the 'code' in a variable (class property) so you can change it easily in the future.